packages/nx/src/project-graph/plugins/plugin-pool.ts

```diff
@@ -51,15 +51,20 @@ export function loadRemoteNxPlugin(plugin: PluginConfiguration, root: string) {
 
   // logger.verbose(`[plugin-worker] started worker: ${worker.pid}`);
 
-  return new Promise<RemotePlugin>((res, rej) => {
-    worker.on('message', createWorkerHandler(worker, res, rej));
-    worker.on('exit', createWorkerExitHandler(worker));
-  });
+  return [
+    new Promise<RemotePlugin>((res, rej) => {
+      worker.on('message', createWorkerHandler(worker, res, rej));
+      worker.on('exit', createWorkerExitHandler(worker));
+    }),
+    () => {
+      shutdownPluginWorkers();
+    },
+  ] as const;
```

Comment:
I feel like this should be an object so we have more well defined labels (e.g.)

```
return {
  plugin: RemotePlugin
  shutdown: () => Promise<void>
}
```

Reply:
Mm.. Idk how much of a difference it makes. I'm going to keep it a tuple for the interest of time.